src/handler/nicknameHandler.ts

```diff
@@ -7,25 +7,39 @@ import type { BotContext } from "../context.js";
 export default class NicknameHandler {
     constructor(private readonly context: BotContext) {}
 
+    private nicknameFromId(id: string): string {
+        return this.context.client.users.cache.get(id)?.username ?? "";
+    }
+
     async rerollNicknames() {
         console.log("rerolling nicknames");
         const allUsersAndNames = Object.entries(
             await Nicknames.allUsersAndNames()
         );
 
-        const updateTasks = allUsersAndNames.map(([userId, nicknames]) => this.updateNickname(userId, nicknames));
+        const updateTasks = allUsersAndNames.map(([userId, nicknames]) => {
+            const thisNickname = this.nicknameFromId(userId);
+            return this.updateNickname(userId, nicknames, thisNickname);
+        });
         await Promise.all(updateTasks);
     }
 
-    async updateNickname(userId: string, storedNicknames: string[]) {
+    private pickUnusedNickname(current: string, nicknames: string[]): string[] {
+        return nicknames.filter((candidate, _index, _arr) => {
+            return candidate !== current;
+        });
+    }
+
+    async updateNickname(userId: string, storedNicknames: string[], current: string) {
         try {
             const member = this.context.guild.members.cache.find(
                 m => m.id === userId
             );
             if (!member) return;
             const nicknames = [member.user.username, ...storedNicknames];
```

Comment:
Du hätest auch nur diese Zeile ändern können. Kann man vielleicht in ne Funktion auslagern, aber deine Änderungen finde ich zu komplex für das Problem.

```suggestion
            const nicknames = [member.user.username, ...storedNicknames];
            const pickableNicknames = nicknames.filter(n => n !== member.nickname);
```

Reply:
Joa das ist viel besser.